### PR TITLE
docs: add the design of table attributes

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: ref https://github.com/tikv/pd/issues/3839

Problem Summary: This PR adds the design of region label in SQL.

### What is changed and how it works?

Only design document.

### Check List

Tests 

- [x] No code

### Release note

```release-note
None
```

### CODE REVIEW

**Review by nolouch — APPROVED** [review]

File: docs/design/2021-10-20-table-attributes.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+### Display Attributes
+
+we add a new table in the `INFORMATION_SCHEMA` named `ATTRIBUTES`. If necessary, we can also support syntax like
+
+```sql
+SHOW ATTRIBUTES FOR [{DATABASE|SCHEMA} s][TABLE t][PARTITION p];
+```
+or 
+```
+SHOW CREATE TABLE;
```
Comment (kennytm) [inline docs/design/2021-10-20-table-attributes.md:55]:
```suggestion
SHOW CREATE TABLE t;
```

?

File: docs/design/2021-10-20-table-attributes.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+| schema/test/t1/p0 | key-range | "key=value"          | [7480000000000000ff3a5f720000000000fa, 7480000000000000ff3b5f720000000000fa]                                                                                                                                                             |
+| schema/test/t1    | key-range | "key=value"          | [7480000000000000ff395f720000000000fa, 7480000000000000ff3a5f720000000000fa], [7480000000000000ff3a5f720000000000fa, 7480000000000000ff3b5f720000000000fa], [7480000000000000ff3b5f720000000000fa, 7480000000000000ff3c5f720000000000fa] |
++-------------------+-----------+----------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+```
+
+### DDL Management
+Here are the concrete behaviors when doing the DDL operations with a table/partition having the attributes.
+
+#### Drop Clause 
++ For the drop table operation, the attributes will be deleted once the GC has finished
```
Comment (kennytm) [inline docs/design/2021-10-20-table-attributes.md:74]:
```suggestion
+ For the drop schema and table operations, the attributes will be deleted once the GC has finished
```

File: docs/design/2021-10-20-table-attributes.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
++ For the drop partition operation, the attributes will be deleted immediately
+
+#### Truncate Clause
++ For the truncate table operation, both table range and partition range of attributes will be updated.
++ For the truncate partition operation, the partition range of attributes will be updated.
+
+#### Recover/Flashback Clause
++ The attributes will be not changed once the recover/flashback process success.
```
Comment (kennytm) [inline docs/design/2021-10-20-table-attributes.md:82]:
```suggestion
+ The attributes will be kept as the old values after the recover/flashback process succeed.
```

File: docs/design/2021-10-20-table-attributes.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+### Privilege Management
+
+Privilege management is quite straightforward:
+
+* `ALTER TABLE` statement requires `Alter` privilege
+* `information_schema.attributes` only shows the attributes on the objects that visible to the current user
+* `ADMIN SHOW DDL` requires `Super` privilege
+
+### Inheritance
+Consider that if we add attributes for a table, all partitions of that table will inherit the attributes by default.
```
Comment (kennytm) [inline docs/design/2021-10-20-table-attributes.md:96]: clarify the behavior of `CREATE TABLE x LIKE y`?

**Review by kennytm — COMMENTED** [review]
i think we need a `/*T! */` feature like placement rules (https://github.com/pingcap/tidb/blob/master/docs/design/2020-06-24-placement-rules-in-sql.md)

rest lgtm.

File: docs/design/2021-10-20-table-attributes.md (hunk 4 above)
Comment (rleungx) [reply docs/design/2021-10-20-table-attributes.md:96]: For now, `x` won't have the same attributes as `y`.

**Comment by rleungx** [comment]
> i think we need a `/*T! */` feature like placement rules (https://github.com/pingcap/tidb/blob/master/docs/design/2020-06-24-placement-rules-in-sql.md)
> 
> rest lgtm.

Only create table clause has `/*T! */` feature in placement rules. Alter clause doesn't have it.

File: docs/design/2021-10-20-table-attributes.md (hunk 4 above)
Comment (kennytm) [reply docs/design/2021-10-20-table-attributes.md:96]: yes i mean this is not written in the document. people shouldn't need to dig discussions around to know that.

Comment (rleungx) [reply docs/design/2021-10-20-table-attributes.md:96]: got

File: docs/design/2021-10-20-table-attributes.md (hunk 2 above)
Comment (rleungx) [reply docs/design/2021-10-20-table-attributes.md:74]: Actually, when dropping schema, the attributes will be deleted immediately.

**Comment by kennytm** [comment]
@rleungx ❓

**Comment by rleungx** [comment]
> @rleungx ❓

I replied the question: https://github.com/pingcap/tidb/pull/28971#discussion_r743347080

File: docs/design/2021-10-20-table-attributes.md (hunk 2 above)
Comment (kennytm) [reply docs/design/2021-10-20-table-attributes.md:74]: @rleungx well in that case update line 75 😓 

i think the attributes of the schema will be deleted immediately, but the attributes of the tables inside the schemas will need to wait for GC.

Comment (rleungx) [reply docs/design/2021-10-20-table-attributes.md:74]: updated

**Review by kennytm — APPROVED** [review]

**Pull request merged**